### PR TITLE
Refactoring of event names and NatSpec comments

### PR DESCRIPTION
**Pull request by Michael-A-Heuer**
## Description

- Added missing NatSpec formatting
- Changed events name to comply with the NatSpec naming convention for events (Object + verb in past participle form)
- Added missing @params in the method's signature
- Modified double comments `//` for triple comments `///`
- Improved voting tests by expecting the `VoteExecuted` event to be emitted

Task: [APP-268](https://aragonassociation.atlassian.net/browse/APP-268), [APP-66](https://aragonassociation.atlassian.net/browse/APP-66)

The work was done together with @jpaulet.

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

## Checklist:

- [x] I have selected the correct base branch.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] Any dependent changes have been merged and published in downstream modules.
- [x] I ran all tests with success and extended them if necessary.
- [x] I have updated the ``CHANGELOG.md`` file in the root folder.
- [x] I have tested my code on the test network.

### CODE REVIEW

File: packages/contracts/contracts/core/DAO.sol [hunk 1]
```diff
@@ -103,7 +103,12 @@ contract DAO is IDAO, Initializable, UUPSUpgradeable, ACL, ERC1271, AdaptiveERC1
         _setMetadata(_metadata);
     }
 
-    /// @inheritdoc IDAO
```
Comment (Rekard0) [inline packages/contracts/contracts/core/DAO.sol:106]: i think we should use @inheritdoc whenever we can

File: packages/contracts/contracts/core/IDAO.sol [hunk 2]
```diff
@@ -31,72 +31,97 @@ abstract contract IDAO {
...
     function deposit(
         address _token,
         uint256 _amount,
         string calldata _reference
     ) external payable virtual;
 
+    /// @notice Emitted when a deposit is made
+    /// @param sender The address of the sender
+    /// @param token The address of the token deposited
+    /// @param amount The amount of tokens deposited
```
Comment (Rekard0) [inline packages/contracts/contracts/core/IDAO.sol:73]: feels like an unfinished sentences
perhaps: `/// @param amount The amount of tokens deposited in the DAO`
or: `/// @param amount The amount of deposited tokens`

same for amount

File: packages/contracts/contracts/core/DAO.sol [hunk 3]
```diff
@@ -196,10 +196,14 @@ contract DAO is IDAO, Initializable, UUPSUpgradeable, ACL, ERC1271, AdaptiveERC1
         _handleCallback(msg.sig, msg.data); // WARN: does a low-level return, any code below would be unreacheable
     }
 
+    /// @notice Emits the MetadataSet event if new metadata is set
+    /// @param _metadata Hash of the IPFS file object
```
Comment (Rekard0) [inline packages/contracts/contracts/core/DAO.sol:200]: do we need `file` ?
`... IPFS object` might be enough.

File: packages/contracts/contracts/votings/whitelist/WhitelistVoting.sol [hunk 4]
```diff
@@ -26,8 +26,9 @@ contract WhitelistVoting is MajorityVoting {
 
     error VoteCreationForbidden(address sender);
 
-    event AddUsers(address[] users);
-    event RemoveUsers(address[] users);
+    event UsersAdded(address[] users);
+
+    event UsersRemoved(address[] users);
```
Comment (Rekard0) [inline packages/contracts/contracts/votings/whitelist/WhitelistVoting.sol:31]: do we need to add NatSpec here?
if yes, the re-visit this file for events and errors

File: packages/contracts/test/adaptive-erc165.ts [hunk 5]
```diff
@@ -102,4 +102,4 @@ describe('AdaptiveErc165', function () {
       expect(await adaptive.supportsInterface('0xabababab')).to.equal(false);
     });
   });
-});
+});
```
Comment (Rekard0) [inline packages/contracts/test/adaptive-erc165.ts:105]: No newline at end of file

Comment (Michael-A-Heuer) [reply packages/contracts/test/adaptive-erc165.ts:105]: My git ignores the whitespace, even with prettier applied. Btw, let's use GitHub suggestions where applicable.
```suggestion
});

```

File: packages/contracts/contracts/core/DAO.sol (hunk 3 above)
Comment (Michael-A-Heuer) [reply packages/contracts/contracts/core/DAO.sol:200]:
```suggestion
    /// @param _metadata Hash of the IPFS metadata object
```

File: packages/contracts/contracts/core/IDAO.sol (hunk 2 above)
Comment (Michael-A-Heuer) [reply packages/contracts/contracts/core/IDAO.sol:73]: I will change it to `/// @notice Emitted when a deposit is made to the DAO`. This way, it does not need to be included twice in the `/// @param` comments

**Review by jpaulet — APPROVED** [review]

**Review by Rekard0 — DISMISSED** [review]

**Review by jpaulet — DISMISSED** [review]

**Review by brickpop — APPROVED** [review]

File: packages/contracts/contracts/votings/ERC20/ERC20Voting.sol [hunk 6]
```diff
@@ -74,7 +71,7 @@ contract ERC20Voting is MajorityVoting {
         uint64 snapshotBlock = getBlockNumber64() - 1;
 
         uint256 votingPower = votingToken.getPastTotalSupply(snapshotBlock);
-        if (votingPower == 0) revert VotePowerZero();
+        if (votingPower == 0) revert VotingPowerZero();
```
[the remark this answers is not in the document]
Comment (brickpop) [reply packages/contracts/contracts/votings/ERC20/ERC20Voting.sol:74]: This is a very minor suggestion, can be ignored and merged (after the CI/CD PR)

**Review by jpaulet — APPROVED** [review]

**Review by jpaulet — APPROVED** [review]

**Pull request merged**